### PR TITLE
feat/splitbutton

### PR DESCRIPTION
**Pull request by elizeumatheus**
image
Kapture 2020-06-17 at 9 59 35

### CODE REVIEW

File: packages/vue-ds/src/components/SplitButton/SplitButton.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+  // it('should have "show" as classname when click on button',  () => {
+  //   const { getByRole } = render(SplitButton, {
+  //     props: {
+  //       text: "Button",
+  //     },
+  //     slots: {
+  //       default: '<a>Action</a>'
+  //     }
+  //   })
```
Comment (tassioFront) [inline packages/vue-ds/src/components/SplitButton/SplitButton.spec.tsx:126]: 👀 hahahaa

File: packages/vue-ds/src/components/SplitButton/SplitButton.vue [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<template>
+  <nav class="SplitButton">
+    <Button :color="color" :size="size" @click="onClickButton">
```
Comment (tassioFront) [inline packages/vue-ds/src/components/SplitButton/SplitButton.vue:3]: I saw that you put `aria-expanded` at React component. At Vue, you can use dynamic props using that strategy: 
```JS
<Button
   v-bind="{'aria-expanded': isOpen}"></Button>
```

File: packages/core/models/index.ts [hunk 3]
```diff
@@ -2,3 +2,4 @@ export * from './button.model'
 export * from './dynamic-components.model'
 export * from './spinner.model'
 export * from './stepper.model'
+export * from './splitbutton.model'
```
Comment (felipefialho) [inline packages/core/models/index.ts:5]: `'./split-button.model'` is better for this file 😊

File: packages/react-ds/src/components/SplitButton/SplitButton.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      >
+        {text}
+        <span className={styles.caret}></span>
+      </Button>
+      <div
+        className={`${styles.dropdown}
+                    ${styles[size]}
+                    ${styles[color]}
+                    ${isOpen ? styles.show : styles.hide}`}
+        role="menu"
```
Comment (felipefialho) [inline packages/react-ds/src/components/SplitButton/SplitButton.tsx:38]: ❤️

File: packages/styles/components/SplitButton.module.scss [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+  &::before,
+  &::after {
+    background-color: var(--white);
+    border-radius: 1px;
+    content: "";
+    display: block;
+    height: 10px;
+    position: absolute;
+    top: 25%;
```
Comment (felipefialho) [inline packages/styles/components/SplitButton.module.scss:20]: Why 25%? 👀

File: packages/styles/components/SplitButton.module.scss [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+  @include reset();
+
+  align-items: center;
+  border-radius: var(--border-radius);
+  box-shadow: -6px 8px 15px rgba(0, 0, 0, 0.1);
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  position: absolute;
+  width: 100%;
```
Comment (felipefialho) [inline packages/styles/components/SplitButton.module.scss:45]: The problem's here is in a context of text of `button` is small that `dropdown` size. Call me that I explain it 😁

File: packages/styles/components/SplitButton.module.scss [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    border: var(--space-xs) solid transparent;
+    content: "";
+    height: 0;
+    margin-left: -20px;
+    pointer-events: none;
+    position: absolute;
+    right: var(--space-sm);
+    top: -16px;
+    width: 0;
+  }
```
Comment (felipefialho) [inline packages/styles/components/SplitButton.module.scss:58]: Why `height: 0;` and `width: 0;`?

File: packages/styles/components/SplitButton.module.scss [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    }
+
+    &.medium {
+      top: calc(var(--space-xlg) - var(--space-xs));
+    }
+  }
+
+  &.hide {
+    opacity: 0;
+    top: var(--space-lg);
```
Comment (felipefialho) [inline packages/styles/components/SplitButton.module.scss:98]: We can use `translateY()` here, there is more performance.

File: packages/styles/components/SplitButton.module.scss [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+
+  align-items: center;
+  border-radius: var(--border-radius);
+  box-shadow: -6px 8px 15px rgba(0, 0, 0, 0.1);
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  position: absolute;
+  width: 100%;
+  z-index: var(--zindex-10);
```
Comment (felipefialho) [inline packages/styles/components/SplitButton.module.scss:46]: To improve animation performance we can add `will-change`.

https://developer.mozilla.org/pt-BR/docs/Web/CSS/will-change

File: packages/vue-ds/src/components/SplitButton/SplitButton.spec.tsx (hunk 1 above)
Comment (felipefialho) [reply packages/vue-ds/src/components/SplitButton/SplitButton.spec.tsx:126]: 👀

File: packages/vue-ds/src/components/SplitButton/SplitButton.vue (hunk 2 above)
Comment (felipefialho) [reply packages/vue-ds/src/components/SplitButton/SplitButton.vue:3]: You are amazing @elizeulouzeiro

File: packages/react-ds/src/components/Button/Button.tsx [hunk 10]
```diff
@@ -14,6 +14,7 @@ interface IButtonProps extends IButton {
   endIcon?: React.ReactNode | string
   /** React Element | string */
   startIcon?: React.ReactNode | string
+  onClick: () => void
```
Comment (s1doka) [inline packages/react-ds/src/components/Button/Button.tsx:17]: I think we could use this:
```suggestion
  onClick: (event: React.MouseEvent<HTMLButtonElement>): void
```

File: packages/vue-ds/src/components/SplitButton/SplitButton.vue [hunk 11]
```diff
@@ -0,0 +1,56 @@
+<template>
+  <nav class="SplitButton">
+    <Button
+      :color="color"
+      :size="size"
+      @click="onClickButton"
+      v-bind="{ 'aria-expanded': isOpen }"
```
Comment (tassioFront) [inline packages/vue-ds/src/components/SplitButton/SplitButton.vue:7]: 😍😍

**Pull request merged**